sixtyfps_runtime/corelib/model.rs

```diff
@@ -694,45 +688,40 @@ impl<C: RepeatedComponent> PinnedDrop for Repeater<C> {
 }
 
 impl<C: RepeatedComponent + 'static> Repeater<C> {
-    fn model(self: Pin<&Self>) -> ModelHandle<C::Data> {
-        // Safety: Repeater does not implement drop and never let access model as mutable
+    fn model(self: Pin<&Self>) -> SharedModel<C::Data> {
+        // Safety: Repeater does not implement drop and never allows access to model as mutable
         let model = self.project_ref().model;
 
         if model.is_dirty() {
             *self.inner.borrow_mut() = RepeaterInner::default();
             self.is_dirty.set(true);
-            if let ModelHandle(Some(m)) = model.get() {
```

Comment:
Too bad that we loose this optimisation that we won't allocate stuff if the model is empty.
I guess that's fine...
Otherwise you could use pub(crate)  to allow us to access the inner one.
Or we make TryInto Rc<> work

Reply:
Will an empty model ever get dirty?

Reply:
it can grow, yes.

Reply:
I mean, what's an empty model.  a model with row_count = 0  but an actual model can grow.
But a model backed by None cannot change unless the property itself is changed. So no, a model with None cannot become dirty

Reply:
So we do not loose the optimization: `None` will never get dirty, so it will never end up here in the first place.

Reply:
`model` in this case is a `Property<ModelRc>`  It's the property that gets dirty here, not the model.
But i guess we can live with that for now.